pipelines/docs-binaries.yml

```diff
@@ -28,7 +30,6 @@ jobs:
         -OutputDirectory $(Build.ArtifactStagingDirectory)
 
   - task: PublishPipelineArtifact@1
-    condition: always()
```

Comment:
Why do we want to remove this? I thought even when the task fails we might still get logs here in certain cases.

Reply:
I've been running into issues where this breaks the re-run flow: if we happen to fail for a non-deterministic reason (like Unity being temporarily unable to reach the licensing server), it's much easier to rerun the failed jobs than to spin up a manual pipeline run.

The issue is that if we've already published an artifact (even if it's empty, which it often is in the licensing failure case), you can't republish something with the same artifact. That puts the CI run into a permanent failure state unless you run a new manual run, which we tend to avoid due to various reasons.

Reply:
I would probably argue that that would be an ADO bug... But yeah I can see why that is annoying.